code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -109,6 +116,43 @@
 	. += emissive_appearance(icon, "locked", alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/obj/structure/closet/proc/animate_door(closing = FALSE)
+	if(door_anim_time)
+		if(!door_obj) door_obj = new
+		vis_contents |= door_obj
```

Comment:
```suggestion
		vis_contents += door_obj
```
https://hackmd.io/@tgstation/ry6DxGvDL#Performance-of--vs--on-lists

IIRC stuff like `vis_contents` is de-duplicated (internally) before rendering as well